src/main/java/com/coreos/jetcd/ClientBuilder.java

```diff
@@ -165,4 +173,39 @@ public ClientBuilder copy() {
       throw EtcdExceptionFactory.newEtcdException(e);
     }
   }
+
+  private static boolean isValidEndpointFormat(String endpoint) {
+    if (isValidHostAndPort(endpoint)) {
```

Comment:
why is this needed? valid host and port will be checked when URL are constructed.